libqpdf/QPDFParser.cc

```diff
@@ -191,20 +191,20 @@ QPDFParser::parse(bool& empty, bool content_stream)
                 if (content_stream) {
                     object = QPDF_Operator::create(value);
                 } else if (
-                    (value == "R") && (state != st_top) && (size >= 2) &&
-                    (!olist.back().isIndirect()) &&
```

Comment:
Why is it okay to remove the indirect check? I think this was to prevent bogus stuff like 10 0 R 0 R where 10 0 R resolves to an integer. By making olist take a list of object pointers, do we lose the ability to check that?

Reply:
Looking through the code, I'm thinking the previous refactorings have actually made that case impossible or such that it won't show up that way anymore.

Reply:
I thought the check was there to prevent the call to isInteger accidentally resolving an indirect object, which would have crashed the parser/qpdf.

10 0 R 0 R may or may not be interpreted as as an indirect reference depending on whether 10 0 R had been resolved prior to calling parse. I think check for indirect integers ought to go back in.

Reply:
We should make sure there's actually a test for it. That code has probably always been there and was probably not added in response to a bug, so it may not have occurred to me to create a test for it.